save-backend/Backend-API.md

```diff
@@ -0,0 +1,269 @@
+
+```bash
+# Required
+testRootPath='"examples/kotlin-diktat"'
+
+# Optional
+branch='"origin/feature/testing_for_cloud"'
+
+# Optional
+commitHash=null
+```
+
+</details>
+
+#### Configuration for Standard Mode
+
+For Standard Mode you need provide at least the list of test suites,
+which should be involved in testing
+
+<details>
+  <summary>Expand</summary>
+
+```bash
+# Required
+testSuites='["Directory: Chapter 1", "Directory: Chapter2"]'
+
+# Optional
+execCmd=null
+
+# Optional
+# Batch size controls how many files will be processed at the same time.
+batchSize=null
+
+```
+</details>
+
+### API for execution
+
+#### Upload additional files into SAVE Cloud storage
+
+If your tests require additional files for execution, they could be uploaded by
+following `post` request:
+
+```bash
+curl -X POST "${SAVE_CLOUD_URL}/api/v1/files/upload" \
+-H "X-Authorization-Source: ${SAVE_CLOUD_AUTH_SOURCE}" \
+-H "Authorization: ${SAVE_CLOUD_AUTH}" \
+-F "file=@your-file-name"
+```
+
+It will return `json` with metadata about your file, which will be require\
+in the execution request later.
+
+The format will have the following form:
+
+```bash
+{
+  "name": "your-file-name",
+  "uploadedMillis": 1651662834923,
+  "sizeBytes":172,
+  "isExecutable":false
+}
+```
+
+If you already uploaded them into the SAVE Cloud storage, you will need\
+to get the corresponding `json`. It could be done by following request,\
+which will return the metadata of all files in SAVE Cloud storage:
+
+```bash
+curl -X GET "${SAVE_CLOUD_URL}/api/v1/files/list" \
+-H "X-Authorization-Source: ${SAVE_CLOUD_AUTH_SOURCE}" \
+-H "Authorization: ${SAVE_CLOUD_AUTH}"
+```
+
+#### Execution submission
+
+For execution request you will need full information about your project, which could be
+saved into variable `project` by following request:
+
+```bash
+project=$(curl -X GET "${SAVE_CLOUD_URL}/api/v1/projects/get/organization-name?name=save&organizationName=${organizationName}" \
+-H "X-Authorization-Source: ${SAVE_CLOUD_AUTH_SOURCE}" \
+-H "Authorization: ${SAVE_CLOUD_AUTH}")
+```
+Unifying all above, here the examples of how to submit execution, both in
+Git and Standard modes via curl:
+
+<details>
+  <summary>Request for execution submission in Git Mode</summary>
+
+```bash
+curl -X POST "${SAVE_CLOUD_URL}/api/v1/submitExecutionRequest" \
+-H "X-Authorization-Source: ${SAVE_CLOUD_AUTH_SOURCE}" \
+-H "Authorization: ${SAVE_CLOUD_AUTH}" \
+-F "executionRequest={
+    \"project\": ${project},
+    \"gitDto\": {
+        \"url\": ${gitUrl},
+        \"username\": ${gitUserName},
+        \"password\": ${gitPassword},
+        \"branch\": ${branch},
+        \"hash\": ${commitHash}
+    },
+    \"testRootPath\": ${testRootPath},
+    \"sdk\": {
+      \"name\": ${sdkName},
+      \"version\": ${sdkVersion}
+    },
+    \"executionId\" : null
+};type=application/json" \
+-F 'file={
+  "name": "ktlint",
+  "uploadedMillis": 1637658398621,
```

Comment:
This is very bad…

Reply:
Add new task in https://github.com/analysis-dev/save-cloud/issues/209